src/Http/Authentication.Core/src/AuthenticationService.cs

```diff
@@ -77,8 +77,13 @@ public virtual async Task<AuthenticateResult> AuthenticateAsync(HttpContext cont
             var result = await handler.AuthenticateAsync();
             if (result != null && result.Succeeded)
             {
-                var transformed = await Transform.TransformAsync(result.Principal);
-                return AuthenticateResult.Success(new AuthenticationTicket(transformed, result.Properties, result.Ticket.AuthenticationScheme));
+                var principal = result.Principal;
+                if (!Options.ApplyClaimsTransformationOnce || result.Properties.ClaimsTransformed != true)
```

Comment:
Using AuthProperties to store this might work too well. E.g. An automatic or manual cookie refresh would preserve the flag and transformation would never run again.
We might need to use an Items entry keyed by some transformation prefix and the auth scheme.

Reply:
We could also just explicitly clear the flag in cases where we are copying auth properties and want transformation to run again, I'd imagine we don't do this in all that many places

Reply:
copying? The coping I referred to is implicit when you serialize-deserialize across requests. Seralization wouldn't/shouldn't know anything about specific values.

Reply:
Right, this is a property we don't want to persist is what I meant.

Reply:
Doesn't seem that unreasonable if we just hard block this property in our properties serializer https://github.com/aspnet/AspNetCore/blob/master/src/Security/Authentication/Core/src/PropertiesSerializer.cs#L55

Reply:
That's a layering disaster waiting to happen... Why not just move to Items?

Reply:
Yeah I guess that gives us the desired effect as well

Reply:
So there's actually a subtle difference by doing this in Items vs something in the ticket itself.  We are making an assumption that the handler is not returning a different authenticate result every time we call authenticate if we track it in items as opposed to stamping the results itself.  

Invoking claims transformation once per ticket seems more correct than just once per request per scheme, but I guess if we name it explicitly `ApplyClaimsTransformationOncePerScheme` that might help

Reply:
How does once per scheme apply in the case of forwarders?

Reply:
Its been a while, so stepping back, the goal is basically to only run claims transformation once per claims principal/ticket instance right?  Can we just use DI to help us and have a scoped service that has a hashmap to track whether we've transformed a particular instance before?  That seems safer than mucking with the ticket itself as the service would just automatically go away every request.

Reply:
Hashmap? How many principals/tickets per request are you expecting? The most complex scnenario I can think of would produce two, one for a client cert and one for a cookie/JWT.

Reply:
Well identity alone has 3 cookies (Application, External, TFA), if the app also uses JWT, that's 4.  But regardless the main idea is just to outsource the tracking into a scoped service, rather than trying to mark the ticket to avoid any potential serialization issues (using httpcontext.items is effectively just using that dictionary as the scoped service)

Reply:
Looks like the authentication service itself is scoped by default, so we can just track the principals we've transformed directly, that pretty easily lets us only apply it once per principal per request